python_training/sim_city.py

```diff
@@ -0,0 +1,207 @@
+
+
+class City:
+    """
+    A city in Ackland.
+    """
+
+    def __init__(self, city_name):
+        """
+        Initiates a new City object.
+
+        :param city_name: The name of the city you would like to create.
+        """
+        self.city_name = city_name
+        self.base_municipality_tax_rate = BASE_MUNICIPALITY_TAX_RATE
+        self.neighbourhoods = {}
+
+    def how_much_money(self) -> int:
+        """
+        Calculates the concluded amount of tax money that would be payed by The cities in Ackland.
+
+        :return: The amount of money that should be payed.
+        """
+        return self.base_municipality_tax_rate + sum([neighbourhood.get_neighbourhood_tax_rate() for neighbourhood
+                                                      in self.neighbourhoods.values()])
+
+    def ruin_a_neighbourhood(self, neighbourhood_name):
+        """
+        Removes a neighbourhood from the city.
+
+        :param neighbourhood_name: The name of the neighbourhood that would be removed.
+        """
+        self._verify_neighbourhood(neighbourhood_name)
+        self.base_municipality_tax_rate *= 1.05
+        self.neighbourhoods.pop(neighbourhood_name)
+
+    def build_a_neighbourhood(self, neighbourhood_name):
+        """
+        Adds a neighbourhood to the city.
+
+        :param neighbourhood_name: The name of the neighbourhood.
+        :return A neighbourhood object if succeeded, None otherwise.
+        """
+        self._verify_neighbourhood(neighbourhood_name, should_exist=False)
+        self.base_municipality_tax_rate *= 1.1
+        self.neighbourhoods[neighbourhood_name] = Neighbourhood(neighbourhood_name, [])
+        return self.neighbourhoods[neighbourhood_name]
+
+    def build_a_house(self, neighbourhood_name, number_of_family_members, size_of_house):
+        """
+        Adds a house to a neighbourhood.
+
+        :param neighbourhood_name: The neighbourhood that a house would be added to.
+        :param number_of_family_members: The number of family members in the house.
+        :param size_of_house: The size of the house.
+        """
+        self._verify_neighbourhood(neighbourhood_name)
+        self.neighbourhoods[neighbourhood_name].add_house(number_of_family_members, size_of_house)
+
+    def build_a_park(self, neighbourhood_name):
+        """
+        Adds a park to a neighbourhood.
+
+        :param neighbourhood_name: The name of the neighbourhood to which the park would be added.
+        """
+        self._verify_neighbourhood(neighbourhood_name)
+        self.neighbourhoods[neighbourhood_name].add_park()
+
+    def _verify_neighbourhood(self, neighbourhood_name, should_exist=True):
+        """
+        Verifies that the neigborhood exists if should_exist= True, verifies that it doesnt exist otherwise.
+        :param should_exist:
+        :param neighbourhood_name: The neighbourhood name that would be verified.
+        """
+        if should_exist:
```

Comment:
This method could be shorter.
No need for the `return` parts when the alternative is to raise. Just use `not`.
For example:
```
if neighbourhood_name not in self.neighbourhoods:
    raise KeyError(...)
if ...:
    raise ...
```